core/src/main/java/org/elasticsearch/common/settings/IndexScopedSettings.java

```diff
@@ -133,8 +134,15 @@
         FsDirectoryService.INDEX_LOCK_FACTOR_SETTING,
         EngineConfig.INDEX_CODEC_SETTING,
         IndexWarmer.INDEX_NORMS_LOADING_SETTING,
-        // this sucks but we can't really validate all the analyzers/similarity in here
-        Setting.groupSetting("index.similarity.", false, Setting.Scope.INDEX), // this allows similarity settings to be passed
+        // validate that built-in similarities don't get redefined
+        Setting.groupSetting("index.similarity.", false, Setting.Scope.INDEX, (s) -> {
+          boolean valid = true;
+          String similarityName = s.substring(0, s.indexOf("."));
+          if(SimilarityService.BUILT_IN.keySet().contains(similarityName)) {
+            throw new IllegalArgumentException("Cannot redefine built-in Similarity [" + similarityName + "]");
+          }
+          return valid;
+        }), // this allows similarity settings to be passed
```

Comment:
this looks great! Can you fix the indentation to 4 spaces and also assert that the key has a `.` in it ie `s.indexOf(".") > 1`  I think it's ready then. Thanks for fixing this great job figuring it out!